### PR TITLE
feat: [SRM-13222]: Show exact error message on Stepper component

### PR DESCRIPTION
**Pull request by deepesh-ui**
### Summary

1. Updated Stepper to show exact errors
2. Updated consumption table to sync with correct time period

#### Screenshots

Screenshot 2022-12-05 at 9 56 56 AM

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by praneshg239 — CHANGES_REQUESTED** [review]
Please check the comments

File: src/modules/85-cv/components/CVStepper/components/Step/Step.tsx [hunk 1]
```diff
@@ -56,8 +56,13 @@ const Step = ({
   )
 
   const stepTitleStatus = useMemo(
-    () => (runValidationOnMount ? getStepStatus(!!isStepValid?.(step.id)) : stepStatus),
-    [isStepValid, runValidationOnMount, step.id, stepStatus]
+    () =>
+      runValidationOnMount
```
[the remark this answers is not in the document]
Comment (deepesh-ui) [reply src/modules/85-cv/components/CVStepper/components/Step/Step.tsx]: done, on readability we can discuss

**Comment by deepesh-ui** [comment]
retrigger jest

**Comment by deepesh-ui** [comment]
retrigger coverage

**Comment by deepesh-ui** [comment]
retrigger titlecheck

**Comment by deepesh-ui** [comment]
retrigger titlecheck

File: cypress/integration/85-cv/slos/CreateCompositeSLO.spec.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ describe('Create Composite SLO', () => {
     cy.visitChangeIntelligenceForSLOs()
   })
 
-  it('should show all validations.', () => {
+  it.only('should show all validations.', () => {
```
Comment (harshilbits) [inline cypress/integration/85-cv/slos/CreateCompositeSLO.spec.ts:48]: please remove only

File: src/modules/85-cv/components/CVStepper/components/Step/Step.tsx [hunk 3]
```diff
@@ -88,9 +95,21 @@ const Step = ({
         {(isCurrentStep || isErrorMessageVisible) && (
           <Container>
             {isErrorMessageVisible && (
-              <Text margin={{ bottom: isCurrentStep ? 'large' : '' }} intent="danger">
-                {getString('cv.CVStepper.StepError')}
-              </Text>
+              <>
+                {!step.errorMessage?.length ? (
```
Comment (harshilbits) [inline src/modules/85-cv/components/CVStepper/components/Step/Step.tsx:99]: null check for errorMessage

File: src/modules/85-cv/pages/slos/CVSLODetailsPage/DetailsPanel/views/CompositeSLOConsumption/CompositeSLOConsumption.utils.tsx [hunk 4]
```diff
@@ -84,31 +84,54 @@ export const RenderAssignedWeightage: Renderer<CellProps<any>> = ({ row }) => {
...
+
+  return [daysFormatted, hoursFormatted, minutesFormatted, secondsFormatted].join('')
+}
+
 export const RenderErrorBudgetBurned: Renderer<CellProps<any>> = ({ row }) => {
   const slo = row.original
   return (
     <Text className={css.titleInSloTable} font={{ align: 'left', size: 'normal', weight: 'semi-bold' }}>
-      {slo?.errorBudgetBurned}
+      {durationAsString(slo?.errorBudgetBurned)}
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/CVSLODetailsPage/DetailsPanel/views/CompositeSLOConsumption/CompositeSLOConsumption.utils.tsx:126]: destructure fields from slo

File: src/modules/85-cv/components/ChangeTimeline/components/TimelineSlider/TimelineSlider.tsx [hunk 5]
```diff
@@ -128,7 +128,7 @@ const TimelineSlider: React.FC<TimelineSliderProps> = ({
       const endX = Math.min(containerWidth, initialSliderWidth + offset)
       const startXPercentage = startX / containerWidth
       const endXPercentage = endX / containerWidth
-      if (startXPercentage && endXPercentage) {
+      if (startXPercentage && endXPercentage && !hideSlider) {
```
[the remark this answers is not in the document]
Comment (deepesh-ui) [reply src/modules/85-cv/components/ChangeTimeline/components/TimelineSlider/TimelineSlider.tsx:131]: this was causing multiple renders got introduced as part of ticket SRM-11364

File: src/modules/85-cv/pages/slos/CVSLODetailsPage/DetailsPanel/views/CompositeSLOConsumption/CompositeSLOConsumption.utils.tsx (hunk 4 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/CVSLODetailsPage/DetailsPanel/views/CompositeSLOConsumption/CompositeSLOConsumption.utils.tsx:126]: being used only once , so didn't destructure

File: src/modules/85-cv/components/CVStepper/components/Step/Step.tsx (hunk 3 above)
Comment (deepesh-ui) [reply src/modules/85-cv/components/CVStepper/components/Step/Step.tsx:99]: step will never be undefined , errorMessage might be undefined  i.e so !_undefined_?.length, which will not break

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/CreateCompositeSloForm.utils.ts [hunk 6]
```diff
@@ -106,6 +118,53 @@ export const isFormDataValid = (
...
+const errorDefineSLOSection = (formikProps: FormikProps<SLOV2Form>) => {
+  const {
+    errors: { name, identifier, userJourneyRef }
+  } = formikProps
+  return [name, identifier, userJourneyRef as string].filter(item => Boolean(item)) as string[]
+}
+
+const errorSetSLOTimeWindow = (formikProps: FormikProps<SLOV2Form>) => {
+  const {
+    errors: { periodLength, periodLengthType, periodType }
```
[the remark this answers is not in the document]
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/CreateCompositeSloForm.utils.ts:130]: done

**Comment by deepesh-ui** [comment]
retrigger jest

File: src/modules/85-cv/components/CVStepper/components/Step/Step.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (praneshg239) [reply src/modules/85-cv/components/CVStepper/components/Step/Step.tsx]: @deepesh-ui I am still seeing the same code, not seeing the change

File: cypress/integration/85-cv/slos/CreateCompositeSLO.spec.ts (hunk 2 above)
Comment (deepesh-ui) [reply cypress/integration/85-cv/slos/CreateCompositeSLO.spec.ts]: done

**Review by praneshg239 — APPROVED** [review]

**Pull request merged**